dags/pegs-challenge-submission-dag.py

```diff
@@ -32,25 +41,43 @@
 
 @dag(**dag_config)
 def pegs_challenge_submission_dag():
-    @task.branch()
-    def check_for_new_submissions(**context):
-        """
-        Checks for new submissions
-
-        Args:
-            evaluation_id (str): Evaluation ID for challenge
-        """
+    @task
+    def get_new_submissions(**context) -> list[int]:
         hook = SynapseHook(context["params"]["synapse_conn_id"])
-        if hook.ops.monitor_evaluation_queue(
-            context["params"]["synapse_evaluation_id"]
-        ):
-            return "launch_model2data_workflow"
+        submissions = hook.ops.get_submissions_with_status(
+            context["params"]["view_id"], "RECEIVED"
+        )
+        return submissions
+
+    @task.branch()
+    def update_submission_statuses(submissions: list, **context) -> str:
+        if submissions:
+            hook = SynapseHook(context["params"]["synapse_conn_id"])
+            for submission in submissions:
+                hook.ops.update_submission_status(
+                    submission_id=submission,
+                    submission_status="EVALUATION_IN_PROGRESS",
+                )
+            return "stage_submissions_manifest"
         return "stop_dag"
 
     @task()
     def stop_dag():
         pass
 
+    @task()
+    def stage_submissions_manifest(submissions: list, **context) -> str:
+        s3_hook = S3Hook(
+            aws_conn_id=context["params"]["aws_conn_id"], region_name=REGION_NAME
+        )
+        df = pd.DataFrame({"submission_id": submissions})
+        df.to_csv(FILE_NAME, index=False)
+        s3_hook.load_file(
+            filename=FILE_NAME, key=f"{KEY}/{FILE_NAME}", bucket_name=BUCKET_NAME
+        )
+        os.remove(FILE_NAME)
+        return f"s3://{BUCKET_NAME}/{KEY}/{FILE_NAME}"
+
     @task()
     def launch_model2data_workflow(**context):
         hook = NextflowTowerHook(context["params"]["tower_conn_id"])
```

Comment:
You might also want to pass the Python scripts for validation, scoring, and emailing as parameters to the workflow run, or use the `pegs_challenge` profile just so its clear exactly what the workflow run is doing when the DAG triggers it.

Reply:
Right now the profile is configured to the real eval queue (not test) so I've given the individual parameters

Reply:
Noted: Once we're ready to go to prod, I won't be using parameters and will just be using the profile.